src/Workspaces/CoreTest/SolutionTests/SolutionTests.cs

```diff
@@ -1811,6 +1811,57 @@ public async Task TestUpdateRootStillCarriesDiagnosticData([CombinatorialValues(
             Assert.True(finalCompilation.ContainsSyntaxTree(syntaxTreeAfterUpdateRoot));
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.Workspace)]
+        [WorkItem(3705, "https://github.com/dotnet/roslyn/issues/3705")]
+        public async Task TestAddingEditorConfigFileWithIsGeneratedCodeOption()
```

Comment:
@jasonmalinowski I have rewritten the test.

Reply:
Love it now. It's doing a pretty good end-to-end test and since it's also changing out the option for a tree ensures we aren't caching the generated state anywhere.